### PR TITLE
feat: use pack-up to build admin

### PR DESCRIPTION
**Pull request by Marc-Roig**
### What does it do?

- Uses pack up to build admin package.
- License info endpoint is now being overwritten in the ee folder, so ce code doesn't need to use strapi/strapi ee  code.

After I migrated to pack up, I realized the merge of controllers and services were not being done properly. Then I realized we need to export them as defaults if we want them to be merged properly.

I am updating the docs so this is mentitoned somewhere.

### CODE REVIEW

File: packages/core/admin/ee/server/src/controllers/admin.ts [hunk 1]
```diff
@@ -6,6 +6,16 @@ import { env } from '@strapi/utils';
...
+  // NOTE: Overrides CE admin controller
+  async getProjectType() {
+    const flags = strapi.config.get('admin.flags', {});
+    try {
+      return { data: { isEE: strapi.EE, features: ee.features.list(), flags } };
+    } catch (err) {
+      return { data: { isEE: false, features: [], flags } };
+    }
+  },
+
```
Comment (Marc-Roig) [inline packages/core/admin/ee/server/src/controllers/admin.ts:18]: I am not completely sure if this is the right solution, but if it is, we will be able to completely avoid the circular dependency with strapi/strapi.
I am investigating now

**Comment by Marc-Roig** [comment]
aaaah, nevermind , we still have the circular dependency issue. pack up builds the strapi/admin first, both ce and ee.
image

I will find another alternative

**Comment by joshuaellis** [comment]
> aaaah, nevermind , we still have the circular dependency issue. pack up builds the strapi/admin first, both ce and ee. image
> 
> I will find another alternative

could you just not generate types?

**Comment by Marc-Roig** [comment]
@joshuaellis , how could we do this?

At the moment my approach was this [`b480a93` (#18582)](https://github.com/strapi/strapi/pull/18582/commits/b480a93b7a04d8dab9ffc09d582731628a2c4416#diff-52ebb7ab4fdbbcf3ebd176fcef3aa9cf674022b3e746426ec2e08fa19b122ebbR249). I would give a kidney for other alternatives

File: packages/core/admin/packup.config.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    //   types: './dist/admin/src/index.d.ts',
+    //   tsconfig: './admin/tsconfig.build.json',
+    //   runtime: 'web',
+    // },
+    {
+      source: './server/src/index.ts',
+      import: './dist/server/index.mjs',
+      require: './dist/server/index.js',
+      types: './dist/server/src/index.d.ts',
+      tsconfig: './server/tsconfig.build.json',
```
Comment (joshuaellis) [inline packages/core/admin/packup.config.ts:18]: remove this

File: packages/core/admin/packup.config.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+      runtime: 'node',
+    },
+    {
+      source: './ee/server/src/index.ts',
+      import: './dist/ee/server/index.mjs',
+      require: './dist/ee/server/index.js',
+      types: './dist/ee/server/src/index.d.ts',
+      tsconfig: './ee/server/tsconfig.build.json',
```
Comment (joshuaellis) [inline packages/core/admin/packup.config.ts:26]: and this, and you won't generate types.

Comment (Marc-Roig) [reply packages/core/admin/packup.config.ts:26]: thank you ! 🚀  will go with that

**Comment by Marc-Roig** [comment]
@jhoward1994 this should be ready for review! circular dependency should be good now, the ee package does not generate types, so it doesn't complain about not strapi/strapi package not being built.

File: packages/core/admin/package.json [hunk 4]
```diff
@@ -30,7 +44,8 @@
     "analyze:deps": "ANALYZE_DEPS=true webpack serve --config webpack.config.dev.js --progress profile",
     "build": "run build:ts && rimraf build && node ./scripts/build.js",
     "build:measure": "run build:ts && rimraf build && cross-env MEASURE_BUILD_SPEED=true node ./scripts/build.js",
-    "build:ts": "run -T tsc -p server/tsconfig.json --outDir ./dist && run build:ts:ee",
+    "build:ts": "pack-up build",
+    "build:ts2": "run -T tsc -p server/tsconfig.json --outDir ./dist && run build:ts:ee",
```
Comment (jhoward1994) [inline packages/core/admin/package.json:48]: Should we remove this script?

Comment (Marc-Roig) [reply packages/core/admin/package.json:48]: oh shit :D yes

**Comment by jhoward1994** [comment]
So am I right in thinking that this will work as long as all the types defined in the ee directory remain internal to that directory? As we are now not generating any ee types in the build? @Marc-Roig

**Comment by Marc-Roig** [comment]
@jhoward1994 exactly, admin ee code should not be used from the outside of this package.

**Review by jhoward1994 — APPROVED** [review]
Makes sense, looks like this approach will work

**Pull request merged**